File: app/src/main/java/com/future/awaker/main/MainActivity.java
package com.future.awaker.main;

import android.content.pm.ActivityInfo;
import android.databinding.DataBindingUtil;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;
import android.view.WindowManager;

import com.future.awaker.R;
import com.future.awaker.databinding.ActivityMainBinding;
import com.future.awaker.home.HomeActivity;

/**
 * Created by ruzhan on 2017/7/12.
 */

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);

        //android O fix bug orientation
        if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        }

        ActivityMainBinding binding =
                DataBindingUtil.setContentView(this, R.layout.activity_main);

        binding.rootRl.post(() -> {
            HomeActivity.launch(this);
            finish();
        });
    }

    @Override
    public void onBackPressed() {

    }
}

File: app/src/main/java/com/future/awaker/MainActivity.kt
package com.future.awaker

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.future.common.CommonUtils
import com.future.common.CommonViewModel

class MainActivity : AppCompatActivity() {

    private val commonViewModel: CommonViewModel by lazy {
        ViewModelProvider(this).get(CommonViewModel::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.container)
        if (savedInstanceState == null) {
            CommonUtils.get().setCommonViewModel(commonViewModel)
            initLiveData()
            val mainFragment = MainFragment.newInstance()
            supportFragmentManager
                .beginTransaction()
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out,
                    R.anim.fade_in, R.anim.fade_out)
                .add(R.id.container, mainFragment, "MainFragment")
                .commitAllowingStateLoss()
        }
    }

    private fun initLiveData() {
        commonViewModel.addFragmentLiveData.observe(this,
            Observer { commonModel ->
                RouterHelper.handleCommonModel(commonModel, supportFragmentManager)
            })
    }

    override fun onBackPressed() {
        if (supportFragmentManager.backStackEntryCount > 0) {
            supportFragmentManager.popBackStackImmediate()
        } else {
            super.onBackPressed();
        }
    }
}
